lms/static/js/spec/student_profile/learner_profile_factory_spec.js

```diff
@@ -62,6 +64,148 @@ define(['backbone', 'jquery', 'underscore', 'common/js/spec_helpers/ajax_helpers
                 LearnerProfileHelpers.expectLimitedProfileSectionsAndFieldsToBeRendered(learnerProfileView);
             });
 
+            it("doesn't show the mode toggle if badges are disabled", function() {
+
+                requests = AjaxHelpers.requests(this);
+
+                var context = createProfilePage(true, {accomplishments_shared: false}),
+                    modeToggleView = context.learnerProfileView.modeToggleView;
+
+                LearnerProfileHelpers.expectModeToggleToBeHidden(requests, modeToggleView);
+            });
+
+            it("renders the mode toggle if there are badges", function() {
+
+                requests = AjaxHelpers.requests(this);
+
+                var context = createProfilePage(true, {accomplishments_shared: true}),
+                    modeToggleView = context.learnerProfileView.modeToggleView;
+
+                requests.currentIndex = 1;
```

Comment:
``currentIndex`` is intended to be an internal implementation detail, so you shouldn't be setting it. It also means that there is an unhandled request in the queue, which means that the code might not be behaving correctly. It would be preferable to handle the request in question.

I like to have the idiom of first verifying that the request was as expected, and then following up immediately with sending a mock response to that request. Something like this:

                AjaxHelpers.expectRequest(requests, 'GET', '/api/team/v0/topics/no_such_topic,course/1', null);
                AjaxHelpers.respondWithError(requests, 404);

https://github.com/edx/edx-platform/blob/master/lms/djangoapps/teams/static/teams/js/spec/views/teams_tab_spec.js#L86

This both consumes the request from the stack, and verifies that it was as expected.